tests/library_wide_test.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+WEB3_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "web3")
+DEFAULT_EXCEPTIONS = (
+    AssertionError,
+    ValueError,
+    AttributeError,
+    TypeError,
+)
+
+
+def test_no_default_exceptions_are_raised_within_web3py():
+    for root, dirs, files in os.walk(WEB3_PATH):
+        if "module_testing" in dirs:
+            dirs.remove("module_testing")
+            continue
+        for file in files:
+            if file.endswith(".py"):
+                file_path = os.path.join(root, file)
+                with open(file_path, "r", encoding="utf-8") as f:
+                    line_number = 0
+                    for line in f:
```

Comment:
Nit: Could use `enumerate` to keep track of the line number instead.

```suggestion
                    for line_number, line in enumerate(f):
```